crates/chain/src/tx_graph.rs

```diff
@@ -699,18 +725,24 @@ impl<A: Anchor> TxGraph<A> {
             }
         };
 
-        // If a conflicting tx is in the best chain, or has `last_seen` higher than this tx, then
-        // this tx cannot exist in the best chain
-        for conflicting_tx in self.walk_conflicts(tx, 25, |_, txid| self.get_tx_node(txid)) {
-            for block in conflicting_tx.anchors.iter().map(A::anchor_block) {
-                if chain.is_block_in_chain(block, chain_tip)? == Some(true) {
-                    // conflicting tx is in best chain, so the current tx cannot be in best chain!
+        // Populate a list of unconfirmed ancestor txs to check for conflicts
+        for tx in TxAncestors::new_include_root(self, tx, |_, tx: &Transaction| {
+            self.get_unconfirmed_tx(tx.txid(), chain, chain_tip)
+                .expect("error is infallible")
```

Comment:
We cannot guarantee that the `ChainOracle` implementation has an [`Infallible`](https://doc.rust-lang.org/std/convert/enum.Infallible.html) error type.

A solution is to have the closure return `Option<Result<&tx, C::Error>>`, and we collect the ancestor txs into `Result<Vec<_>, C::Error>`. You'll need to play around with the lifetimes.